docs/concepts/Package-Versioning.md

```diff
@@ -52,21 +52,24 @@ That said, package developers generally follow recognized naming conventions:
 - `-rc`: Release candidate, typically a release that's potentially final (stable) unless significant bugs emerge.
 
 > [!Note]
-> NuGet 4.3.0+ supports [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html), which supports pre-release numbers with dot notation, as in *1.0.1-build.23*. Dot notation is not supported with NuGet versions before 4.3.0. You can use a form like *1.0.1-build23*.
+> NuGet 4.3.0+ supports [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html), which supports pre-release numbers with dot notation, as in *1.0.1-build.23*. Dot notation is not supported with NuGet versions before 4.3.0. You can use a form like *1.0.1-build23*. It is recommended to use NuGet 4.3.0+, because nuget.org and newer NuGet clients will not treat "1.0.1-build23" as greater than "1.0.1-build3".
 
-When resolving package references and multiple package versions differ only by suffix, NuGet chooses a version without a suffix first, then applies precedence to pre-release versions in reverse alphabetical order. For example, the following versions would be chosen in the exact order shown:
+When resolving package references and multiple package versions differ only by suffix, NuGet chooses a version without a suffix first, then applies precedence to pre-release versions in reverse alphabetical order and treats dot notation numbers with numerical order. For example, the following versions would be chosen in the exact order shown:
```

Comment:
I think this doc needs some more general work. 

For example, this section seems to refer to semver 1.x only, not semver 2.x (cause there's a 2.x section later on). 

Would tabs make it easier? 1.x and 2.x, default to 2.x and then have different explanations for both?

cc @NuGet/nuget-client @NuGet/nuget-pm what do we think?

Reply:
I guess, as a start, we may want to pick a different package than Entity Framework to demonstrate SemVer2.0 - but I am not an expert on Microsoft packages.  I just know that Joel Verhagen explained to me my incorrect application of SemVer2.0.  Alana at ProGet was also helpful in pointing out to me that SemVer 2.0 does not allow four part names, and that while the NuGet Client at various releases may allow this, the standard does not support this.